be/src/exprs/vectorized/string_functions.cpp

```diff
@@ -539,11 +539,17 @@ ColumnPtr string_func_const(StringConstFuncType func, const Columns& columns, Ar
             ColumnPtr binary = func(columns, src_binary, std::forward<Args>(args)...);
             NullColumnPtr src_null = NullColumn::create(*(src_nullable->null_column()));
 
-            // if binary is ConstColumn, just return it.
-            // if binary is NullableColumn, then the null column inside of the binary should
-            // be merged with the null column inside of columns[0].
+            // - if binary is null ConstColumn, just return it.
+            // - if binary is non-null ConstColumn, unfold it and wrap with src_null.
+            // - if binary is NullableColumn, then the null column inside the binary should
+            //   be merged with the null column inside of columns[0].
             if (binary->is_constant()) {
-                return binary;
+                if (binary->only_null()) {
+                    return binary;
+                }
+                ConstColumn* dst_const = down_cast<ConstColumn*>(binary.get());
+                dst_const->data_column()->assign(dst_const->size(), 0);
```

Comment:
only_null returning true indicates that the column is const null.
so break the logic into cases as follows:
1. const null:  only_null()
2. constant column: is_constant()
3. nullable column: is_nullable()
4. elementry column: else

Reply:
Fixed.